pkg/cli/clients/management.go

```diff
@@ -205,136 +220,112 @@ func (amc *UCPApplicationsManagementClient) DeleteResource(ctx context.Context,
-	resourcesWithApplication, err := amc.ListAllResourcesByApplication(ctx, applicationName)
-	if err != nil && !clientv2.Is404Error(err) {
-		return false, err
-	}
-
-	g, groupCtx := errgroup.WithContext(ctx)
-	for _, resource := range resourcesWithApplication {
-		resource := resource
-		g.Go(func() error {
-			_, err := amc.DeleteResource(groupCtx, *resource.Type, *resource.Name)
-			if err != nil {
-				return err
-			}
-			return nil
-		})
-	}
-
-	err = g.Wait()
+// GetApplicationGraph retrieves the application graph of an application by its name (or id).
+func (amc *UCPApplicationsManagementClient) GetApplicationGraph(ctx context.Context, applicationNameOrID string) (corerpv20231001.ApplicationGraphResponse, error) {
+	scope, name, err := amc.extractScopeAndName(applicationNameOrID)
 	if err != nil {
-		return false, err
+		return corerpv20231001.ApplicationGraphResponse{}, err
 	}
 
-	client, err := corerpv20231001.NewApplicationsClient(amc.RootScope, &aztoken.AnonymousCredential{}, amc.ClientOptions)
+	client, err := amc.createApplicationClient(scope)
 	if err != nil {
-		return false, err
+		return corerpv20231001.ApplicationGraphResponse{}, err
 	}
 
-	var respFromCtx *http.Response
-	ctxWithResp := runtime.WithCaptureResponse(ctx, &respFromCtx)
-
-	_, err = client.Delete(ctxWithResp, applicationName, nil)
+	getResponse, err := client.GetGraph(ctx, name, map[string]any{}, &corerpv20231001.ApplicationsClientGetGraphOptions{})
 	if err != nil {
-		return false, err
+		return corerpv20231001.ApplicationGraphResponse{}, err
 	}
 
-	return respFromCtx.StatusCode != 204, nil
+	return getResponse.ApplicationGraphResponse, nil
 }
 
-// CreateOrUpdateApplication creates or updates an application.
-//
+// CreateOrUpdateApplication creates or updates an application by its name (or id).
+func (amc *UCPApplicationsManagementClient) CreateOrUpdateApplication(ctx context.Context, applicationNameOrID string, resource *corerpv20231001.ApplicationResource) error {
+	scope, name, err := amc.extractScopeAndName(applicationNameOrID)
+	if err != nil {
+		return err
+	}
 
-// CreateOrUpdateApplication creates or updates an application resource in Azure using the
-// given application name and resource. It returns an error if the creation or update fails.
-func (amc *UCPApplicationsManagementClient) CreateOrUpdateApplication(ctx context.Context, applicationName string, resource corerpv20231001.ApplicationResource) error {
-	client, err := corerpv20231001.NewApplicationsClient(amc.RootScope, &aztoken.AnonymousCredential{}, amc.ClientOptions)
+	client, err := amc.createApplicationClient(scope)
 	if err != nil {
 		return err
 	}
 
-	_, err = client.CreateOrUpdate(ctx, applicationName, resource, nil)
+	// See: https://github.com/radius-project/radius/issues/7597
+	//
+	// This is a workaround because the server can return invalid system data, which
+	// fails to roundtrip when the client does a "GET -> modify -> PUT".
+	resource.SystemData = nil
```

Comment:
This is a new bug that was exposed while making this change. Fortunately the workaround for it pretty isolated.

Existing functionality in the client code was already impacted by this bug, but we weren't aware of it. With the changes I made to the environment code, it hit the bug in our functional tests!